src/python/tests/core/api/test_scheduler.py

```diff
@@ -0,0 +1,283 @@
+        retval = dict(r.json())
+        assert not retval['schedule']['id'] is None
+
+        # These values did not change
+        assert retval['schedule']['exclusive'] is True
+        # TODO: There is a bug in core/scheduler.py. It does not update the schedule type BUT if you pass a new schedule
+        # type in above, it will return the new schedule type even though it does not update the DB record.
+        assert retval['schedule']['type'] == "INTERVAL"
+        assert retval['schedule']['time'] == "None"
+        assert retval['schedule']['day'] is None
+        assert retval['schedule']['process_name'] == 'testsleep30'
+
+        # Below two values only changed
+        assert retval['schedule']['repeat'] == '0:00:04'
+        assert retval['schedule']['name'] == 'test_update_sch_upd'
+
+    @pytest.mark.run(order=5)
+    @pytest.mark.asyncio
+    async def test_delete_schedule(self):
+        # First create a schedule to get the schedule_id
+        data = {"type": 3, "name": "test_delete_sch", "process_name": "testsleep30", "repeat": "3600"}
+        schedule_id = self._create_schedule(data)
+
+        # Now check the schedules
+        r = requests.delete(BASE_URL+'/schedule/' + schedule_id)
+        retval = dict(r.json())
+
+        assert 200 == r.status_code
+        assert retval['id'] == schedule_id
+        assert retval['message'] == "Schedule deleted successfully"
+
+    @pytest.mark.run(order=6)
+    @pytest.mark.asyncio
+    async def test_get_schedule(self):
+        # First create a schedule to get the schedule_id
+        data = {"type": 3, "name": "test_get_sch", "process_name": "testsleep30", "repeat": "3600"}
+        schedule_id = self._create_schedule(data)
+
+        # Now check the schedule
+        r = requests.get(BASE_URL+'/schedule/' + schedule_id)
+        retval = dict(r.json())
+
+        assert 200 == r.status_code
+        assert retval['id'] == schedule_id
+        assert retval['exclusive'] is True
+        assert retval['type'] == "INTERVAL"
+        assert retval['time'] == "None"
+        assert retval['day'] is None
+        assert retval['process_name'] == 'testsleep30'
+        assert retval['repeat'] == '1:00:00'
+        assert retval['name'] == 'test_get_sch'
+
+    @pytest.mark.run(order=7)
+    @pytest.mark.asyncio
+    async def test_get_schedules(self):
+        # First create two schedules to get the schedule_id
+        data1 = {"type": 3, "name": "test_get_schA", "process_name": "testsleep30", "repeat": "3600"}
+        schedule_id1 = self._create_schedule(data1)
+
+        await asyncio.sleep(4)
+
+        data2 = {"type": 2, "name": "test_get_schB", "process_name": "testsleep30", "day": 5, "time": 44500}
+        schedule_id2 = self._create_schedule(data2)
+
+        await asyncio.sleep(4)
+
+        # Now check the schedules
+        r = requests.get(BASE_URL+'/schedule')
+        retval = dict(r.json())
+
+        list_id = list()
```

Comment:
instead of this, better can be find in retval for schedule_id1 and schedule_id2 and assert against!